x-pack/plugins/security_solution/server/lib/detection_engine/signals/reason_formatters.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { RulesSchema } from '../../../../common/detection_engine/schemas/response/rules_schema';
+import { SignalSourceHit } from './types';
+
+export interface BuildReasonMessageArgs {
+  rule: RulesSchema;
+  mergedDoc?: SignalSourceHit;
+  timestamp: string;
+}
+
+export type BuildReasonMessage = (args: BuildReasonMessageArgs) => string;
+
+/**
+ * Currently all security solution rule types share a commone reason message string. This function composes that string
+ * In the future there may be different configurations based on the different rule types, so the plumbing has been put in place
+ * to more easily allow for this in the future.
+ * @export buildCommonReasonMessage - is only exported for testing purposes, and only used internally here.
+ */
+export const buildCommonReasonMessage = ({
+  rule,
+  mergedDoc,
+  timestamp,
+}: BuildReasonMessageArgs) => {
+  if (!rule) {
+    // This should never happen, but in case, better to not show a malformed string
+    return '';
+  }
+  let hostName;
+  let userName;
+  let timestampForReason = timestamp;
+  if (mergedDoc?.fields) {
+    hostName = mergedDoc.fields['host.name'] != null ? mergedDoc.fields['host.name'] : hostName;
+    userName = mergedDoc.fields['user.name'] != null ? mergedDoc.fields['user.name'] : userName;
+    timestampForReason =
```

Comment:
nit: This ternary operator chain is a bit hard to read. I think that a simple if/else statement might be more readable.

Reply:
Yea, I changed it. Thanks!

Reply:
it should just be the passed in timestamp. So removed all this logic